lib/utils/buildHelper.js

```diff
@@ -0,0 +1,128 @@
+const log = require("@ui5/logger").getLogger("cli:utils:buildHelper");
+
+/**
+ * Creates a list of dependencies as String or RegExp by processing the given <code>dependencies</code> and
+ * <code>dependenciesRegExp</code> parameters. Dependencies are only added to the list if they exist in the
+ * project <code>tree</code>. Optionally, also the sub-dependencies of given <code>dependencies</code> are
+ * added to the list.
+ *
+ * @param {object} parameters Parameters
+ * @param {object} parameters.tree Project tree as generated by the
+ *   [@ui5/project.normalizer]{@link module:@ui5/project.normalizer}
+ * @param {Array.<string>} parameters.dependencies The dependencies to be considered
+ * @param {Array.<string>} [parameters.dependenciesRegExp] The dependencies to be considered as new
+ *   RegExp instances
+ * @param {boolean} [parameters.handleSubtree=false] Decides whether the sub-dependencies of the given
+ *   dependencies are included into the returned dependency list
+ * @returns {Array.<string|RegExp>} Array of the computed dependencies
+ */
+function createDependencyList({tree, dependencies = [], dependenciesRegExp = [], handleSubtree}) {
+	const dependencyList = [];
+	if (handleSubtree && dependenciesRegExp.length) {
+		throw new Error("RegExp's should not be appended to list of sub-dependencies");
```

Comment:
Hard to get this error message when you're not into the details of the method. 

Rather something like "dependenciesRegExp can't be combined with handleSubtree:true option"